test/grep.js

```diff
@@ -174,3 +174,225 @@ test('the pattern looks like an option', t => {
   t.falsy(shell.error());
   t.is(result.toString(), '-v\n-vv\n');
 });
+
+//
+// Before & after contexts
+//
+test('-B option', t => {
+  const result = shell.grep('-B', 3, 'test*', 'test/resources/grep/file3');
+  t.falsy(shell.error());
+  t.is(
+    result.toString(),
+    'line1\n' +
+      'line2 test line\n' +
+      '--\n' +
+      'line1\n' +
+      'line2 test line\n' +
+      'line3 test line\n' +
+      '--\n' +
+      'line7\n' +
+      'line8\n' +
+      'line9\n' +
+      'line10 test line\n' +
+      '--\n' +
+      'line12\n' +
+      'line13\n' +
+      'line14\n' +
+      'line15 test line\n'
+  );
+});
+
+test('-B option, -n option', t => {
+  const result = shell.grep('-nB', 3, 'test*', 'test/resources/grep/file3');
+  t.falsy(shell.error());
+  t.is(
+    result.toString(),
+    '1-line1\n' +
+      '2:line2 test line\n' +
+      '--\n' +
+      '1-line1\n' +
+      '2-line2 test line\n' +
+      '3:line3 test line\n' +
+      '--\n' +
+      '7-line7\n' +
+      '8-line8\n' +
+      '9-line9\n' +
+      '10:line10 test line\n' +
+      '--\n' +
+      '12-line12\n' +
+      '13-line13\n' +
+      '14-line14\n' +
+      '15:line15 test line\n'
+  );
+});
+
+test('-A option, separated', t => {
+  const result = shell.grep('-A', 2, 'test*', 'test/resources/grep/file3');
+  t.falsy(shell.error());
+  t.is(
+    result.toString(),
+    'line2 test line\n' +
+      'line3 test line\n' +
+      'line4\n' +
+      '--\n' +
+      'line3 test line\n' +
+      'line4\n' +
+      'line5\n' +
+      '--\n' +
+      'line10 test line\n' +
+      'line11\n' +
+      'line12\n' +
+      '--\n' +
+      'line15 test line\n\n'
+  );
+});
+
+test('-A option, -n option', t => {
```

Comment:
I think you can omit this case. I think `test('-B option, -n option')` and `test('-A option, separated')` cover this case nicely.

Reply:
Done!